modules/nextflow/src/main/groovy/nextflow/extension/PublishOp.groovy

```diff
@@ -18,47 +18,48 @@ package nextflow.extension
 
 import java.nio.file.Path
 
-import groovy.json.JsonOutput
 import groovy.transform.CompileStatic
 import groovy.util.logging.Slf4j
 import groovyx.gpars.dataflow.DataflowReadChannel
-import nextflow.Global
 import nextflow.Session
+import nextflow.file.FileHelper
 import nextflow.processor.PublishDir
 import nextflow.util.CsvWriter
 /**
- * Publish files from a source channel.
+ * Publish a workflow output.
  *
+ * @author Ben Sherman <bentshermann@gmail.com>
  * @author Paolo Di Tommaso <paolo.ditommaso@gmail.com>
  */
 @Slf4j
 @CompileStatic
 class PublishOp {
 
+    private Session session
+
     private DataflowReadChannel source
 
     private Map opts
 
-    private Path targetDir
+    private String path
 
-    private Closure pathAs
+    private Closure dynamicPath
```

Comment:
Maybe `dynamicResolver` or  `pathResolver` or just `resolver` is not explicit

Reply:
Good point, I was struggling to find a good name for it